internal/cli/login.go

```diff
@@ -239,6 +316,81 @@ func (c *LoginCommand) loginOIDC() (string, int) {
 	return respToken.Token, 0
 }
 
+func (c *LoginCommand) loginK8S(ctx context.Context) (string, int) {
```

Comment:
Adding this makes me think we should have some kind of Interface for our authentication functions here for a more rigid way to define these auth funcs, thoughts? I could see us adding more in the future.

Reply:
Its already typed to `func(context.Context) (string,int)` and in the switch above, so it sort of already is? I don't think we need a separate full Go interface cause they can't stand on their own (they require the login command metadata).

Reply:
Cool cool, makes sense!